src/services/address_service.js

```diff
@@ -57,6 +57,38 @@ export default (easypostClient) =>
       }
     }
 
+    /**
+     * Retrieve a list of all addresses associated with the API key.
+     * @param {object} params
+     * @returns {Address[]}
+     */
+    static async all(params = {}) {
+      try {
+        const url = this._url;
+        const response = await easypostClient.get(url, params);
+
+        return this.convertToEasyPostObject(response.body);
+      } catch (e) {
+        return Promise.reject(e);
+      }
+    }
```

Comment:
This should probably be defined in the base service and we should have an `all` function that implements the base `all` function. See PHP on how we can accomplish this. Much less code and we get the best of both worlds where we have a base function that gets used by individual services.

Reply:
That goes counter to the point of this PR?

Reply:
The purpose of the initiative was that each service had an explicit set of functions available to it without inheriting or needing to not implement each.

The underlying guts don't change for these functions, set a URL, make a request, convert, and reject if an error occurs. Because of that, we should consolidate all that in the base service and each individual service can then implement those few functions and pass in the needed params. We do this with all our other libs.

This would fix the drop in coverage because we wouldn't need to test every individual promise reject line. Doing it this way reduces code duplication while still addressing your initial concern of not implementing certain functions on services.

Reply:
But we'd have to un-implement the "all" function for some services that don't have it.